### PR TITLE
feat(FE): global time visibility as per route

### PR DESCRIPTION
**Pull request by hussamkhatib**
- Moved the function declaration(checkRouteExists) outside component definition 
 - remove for-loop and pass the routes as array in second arg of `matchPath` method,

Fixes: #1659 

Additional Notes
- This check looks unnecessary, this component should never render at `/signup` route as the user is still not [loggedIn](https://github.com/SigNoz/signoz/blob/develop/frontend/src/container/AppLayout/index.tsx#L201).
```
if (history.location.pathname === ROUTES.SIGN_UP) {
		return null;
}
```

### CODE REVIEW

File: frontend/src/container/TopNav/index.tsx [hunk 1]
```diff
@@ -8,38 +8,36 @@ import ShowBreadcrumbs from './Breadcrumbs';
...
-	ROUTES.INSTRUMENTATION,
-	ROUTES.VERSION,
-	ROUTES.ALL_DASHBOARD,
-	ROUTES.ORG_SETTINGS,
-	ROUTES.ERROR_DETAIL,
-	ROUTES.ALERTS_NEW,
-	ROUTES.EDIT_ALERTS,
-	ROUTES.LIST_ALL_ALERT,
+const skipDateTimeSelectorRoutes = [
+	'/settings',
```
Comment (palashgdev) [inline frontend/src/container/TopNav/index.tsx:12]: @hussamkhatib can you pick route from ROUTES object ?

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/TopNav/index.tsx (hunk 1 above)
Comment (hussamkhatib) [reply frontend/src/container/TopNav/index.tsx:12]: yes.
I thought so, But the eg mentioned in issue was done using absolute values of it.


> need to refactor the logic to display according to the route where this component is required
I din't quiet get what the that means

Comment (hussamkhatib) [reply frontend/src/container/TopNav/index.tsx]: Also, rolled back to use the Routes object

**Comment by palashgdev** [comment]
> * Moved the function declaration(checkRouteExists) outside component definition
> * remove for-loop and pass the routes as array in second arg of `matchPath` method,
> 
> Fixes: #1659
> 
> Additional Notes
> 
> * This check looks unnecessary, this component should never render at `/signup` route as the user is still not [loggedIn](https://github.com/SigNoz/signoz/blob/develop/frontend/src/container/AppLayout/index.tsx#L201).
> 
> ```
> if (history.location.pathname === ROUTES.SIGN_UP) {
> 		return null;
> }
> ```

Hi @hussamkhatib 

we need this check 
```
if (history.location.pathname === ROUTES.SIGN_UP) {
		return null;
}
```

File: frontend/src/container/Trace/Filters/Panel/PanelHeading/index.tsx [hunk 2]
```diff
@@ -92,18 +95,10 @@ function PanelHeading(props: PanelHeadingProps): JSX.Element {
 				});
 
 				dispatch({
-					type: UPDATE_ALL_FILTERS,
+					type: UPDATE_FILTER_TO_FETCH_DATA,
```
Comment (palashgdev) [inline frontend/src/container/Trace/Filters/Panel/PanelHeading/index.tsx:98]: can you please revert this one

File: frontend/src/container/Trace/Filters/Panel/PanelHeading/index.tsx [hunk 3]
```diff
@@ -142,21 +137,11 @@ function PanelHeading(props: PanelHeadingProps): JSX.Element {
 			...filterToFetchData.filter((name) => name !== PanelName),
 		];
 
-		// preSelectedFilter.delete(PanelName);
-
 		dispatch({
-			type: UPDATE_ALL_FILTERS,
+			type: UPDATE_FILTER_TO_FETCH_DATA,
```
Comment (palashgdev) [inline frontend/src/container/Trace/Filters/Panel/PanelHeading/index.tsx:141]: same here also

**Comment by hussamkhatib** [comment]
Resolved the issues. The code asked to revert was added unintentionally in the first place (probably because of working in multiple branches).

Btw, I am not sure If PR solves the issue or not. Can you edit the issue to make it more understandable If my PR is not what you are looking for ?

**Pull request closed without merging**